src/api/blob/controller_test.go

```diff
@@ -0,0 +1,200 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package blob
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/goharbor/harbor/src/pkg/distribution"
+	htesting "github.com/goharbor/harbor/src/testing"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/suite"
+)
+
+type ControllerTestSuite struct {
```

Comment:
Seems the testing suite uses the real blob manager directly and the manager uses the real DAO directly, not the faked ones? That means all unit test cases can not run without external database?

Reply:
Ideally, the unit test should only cover its own logic